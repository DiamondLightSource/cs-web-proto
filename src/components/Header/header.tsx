import {
  Border,
  BorderStyle,
  Color,
  DrawerWidget,
  EmbeddedDisplay,
  Font,
  FontStyle,
  OutlineContext,
  RelativePosition
} from "@dls-controls/cs-web-lib";
import React from "react";
import classes from "./header.module.css";

export const Header = (props: { drawer: boolean }): JSX.Element => {
  const { toggleOutlines } = React.useContext(OutlineContext);
  const children = [
    <EmbeddedDisplay
      position={new RelativePosition()}
      key="0"
      file={{ path: "json/drawer.json", defaultProtocol: "pva", macros: {} }}
    />
  ];
  return (
    <header className={classes.header}>
      <div className={classes.imgWrapper}>
        {props.drawer ? (
          <DrawerWidget
            text="Open screens"
            position={new RelativePosition(undefined, "4rem", "1rem")}
            font={new Font(20, FontStyle.Regular)}
            border={
              new Border(
                BorderStyle.Line,
                new Color("var(--diamond-yellow)"),
                3
              )
            }
            foregroundColor={new Color("var(--diamond-yellow)")}
            backgroundColor={new Color("var(--diamond-blue)")}
          >
            {children}
          </DrawerWidget>
        ) : (
          <a href="/">Back to app</a>
        )}
      </div>
      <h1 className={classes.h1}>
        <a href="/">cs-web-proto</a>
      </h1>
      <div className={classes.buttonWrapper}>
        <button
          className={classes.button}
          type="button"
          onClick={toggleOutlines}
        >
          Show outlines
        </button>
      </div>
    </header>
  );
};
