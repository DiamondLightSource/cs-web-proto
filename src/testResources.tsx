import React from "react";
import { DType, DAlarm } from "./types/dtypes";
import { FileProvider, PageState, TabState } from "./fileContext";
import { render, RenderResult } from "@testing-library/react";
import { Provider } from "react-redux";
import { MacroContext } from "./types/macros";
import { csReducer, CsState } from "./redux/csState";
import { createStore } from "redux";
import { BrowserRouter as Router } from "react-router-dom";
import {
  OPEN_WEBPAGE,
  WidgetAction,
  WritePv,
  WRITE_PV
} from "./ui/widgets/widgetActions";

// Helper functions for dtypes.
export function ddouble(
  doubleValue: number,
  alarm: DAlarm = DAlarm.NONE
): DType {
  return new DType({ doubleValue: doubleValue }, alarm);
}

export function ddoubleArray(
  arrayValue: number[],
  alarm: DAlarm = DAlarm.NONE
): DType {
  return new DType({ arrayValue: Float64Array.from(arrayValue) }, alarm);
}

export function dstring(
  stringValue: string,
  alarm: DAlarm = DAlarm.NONE
): DType {
  return new DType({ stringValue: stringValue }, alarm);
}

// Test actions
export const WRITE_PV_ACTION: WritePv = {
  type: WRITE_PV,
  writePvInfo: {
    pvName: "PV",
    value: "value",
    description: "write value to PV"
  }
};

export const WRITE_PV_ACTION_NO_DESC: WritePv = {
  type: WRITE_PV,
  writePvInfo: {
    pvName: "PV",
    value: "value"
  }
};

export const OPEN_BBC_ACTION: WidgetAction = {
  type: OPEN_WEBPAGE,
  openWebpageInfo: { url: "https://bbc.co.uk", description: "BBC" }
};

export const ACTIONS_EX_AS_ONE = {
  actions: [WRITE_PV_ACTION, WRITE_PV_ACTION_NO_DESC],
  executeAsOne: true
};

export const ACTIONS_EX_FIRST = {
  actions: [WRITE_PV_ACTION, WRITE_PV_ACTION_NO_DESC],
  executeAsOne: false
};

// Helper function for rendering with a working fileContext.
export function contextRender(
  component: JSX.Element,
  initialPageState: PageState = {},
  initialTabState: TabState = {},
  initialCsState: CsState = {
    effectivePvNameMap: {},
    globalMacros: {},
    subscriptions: {},
    valueCache: {},
    deviceCache: {}
  }
): RenderResult {
  const ParentComponent = (props: { child: JSX.Element }): JSX.Element => {
    // Hard-code macros for now.
    // eslint-disable-next-line no-template-curly-in-string
    const contextMacros = { a: "A", b: "B", c: "C", e: "${a}" };
    const globalMacros = { c: "D", d: "E" };
    initialCsState.globalMacros = globalMacros;
    const macroContext = { macros: contextMacros, updateMacro: (): void => {} };
    const store = createStore<CsState, any, any, any>(
      csReducer,
      initialCsState
    );
    return (
      <Router>
        <Provider store={store}>
          <MacroContext.Provider value={macroContext}>
            <FileProvider
              initialPageState={initialPageState}
              initialTabState={initialTabState}
            >
              {props.child}
            </FileProvider>
          </MacroContext.Provider>
        </Provider>
      </Router>
    );
  };
  return render(<ParentComponent child={component}></ParentComponent>);
}
