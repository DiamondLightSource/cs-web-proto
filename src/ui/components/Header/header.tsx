import React from "react";
import { OutlineContext } from "../../../outlineContext";
import { ThemeContext } from "../../../themeContext";
import classes from "./header.module.css";

export const Header = (): JSX.Element => {
  const { toggleTheme } = React.useContext(ThemeContext);
  const { toggleOutlines } = React.useContext(OutlineContext);
  return (
    <header className={classes.header}>
      <div className={classes.imgWrapper}>
        <img
          src="img/Diamond_logo_col.jpg"
          alt="Diamond logo"
          className={classes.img}
        ></img>
      </div>
      <h1 className={classes.h1}>cs-web-proto</h1>
      <div className={classes.buttonWrapper}>
        <button type="button" onClick={toggleTheme}>
          Toggle Theme
        </button>
        <button type="button" onClick={toggleOutlines}>
          Show outlines
        </button>
      </div>
    </header>
  );
};
