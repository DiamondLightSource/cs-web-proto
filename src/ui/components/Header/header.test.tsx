import React from "react";
import { contextRender } from "../../../testResources";
import { Header } from "./header";

describe("<Header />", (): void => {
  test("it renders the title", (): void => {
    const { getByText } = contextRender(<Header drawer={true} />);
    expect(getByText("cs-web-proto")).toBeInTheDocument();
  });
});
