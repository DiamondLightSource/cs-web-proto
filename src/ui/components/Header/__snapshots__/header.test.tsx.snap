// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it matches the snapshot 1`] = `
<header
  className="header"
>
  <div
    className="imgWrapper"
  >
    <img
      alt="Diamond logo"
      className="img"
      src="img/Diamond_logo_col.jpg"
    />
  </div>
  <h1
    className="h1"
  >
    cs-web-proto
  </h1>
  <div
    className="buttonWrapper"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Toggle Theme
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Show outlines
    </button>
  </div>
</header>
`;
